### ISSUE TITLE
Move OpenShift related commands into the 'openshift' context

### ISSUE THREAD

**Issue by hferentschik** [minishift/minishift#465]
The introduction of the 'openshift' command (view, set, restart), it might make sense to move some other commands under there as well while we are still in beta state. I am thinking of

* 'service'
* ' get-openshift-versions' (renaming it to 'versions')
* 'logs' - we could keep two logs. One in root for the VM system logs and one for the OpenShift logs under 'openshift'

I am after something like this:
    
    $ minishift openshift -h
    Usage:
      minishift openshift SUBCOMMAND [flags]
      minishift openshift [command]
    
    Available Commands:
      config      Displays or patches OpenShift configuration.
      restart     Restarts the OpenShift cluster.
      versions  Displays Minishift compatible OpenShift versions
      service    Displays info about the specified service (route)
      logs         Displays the OpenShift logs

**Comment by jorgemoralespou** [comment]
@hferentschik I think this might end being too verbose. What commands would not require openshift in the middle? just start/stop/delete? on the other hand, I keep looking into alignment with upstream, so minikube users has the easiest entry path into openshift.

**Comment by hferentschik** [comment]
>  I think this might end being too verbose. 

right, verbosity is an issue here. On the other hand one gains clarity and consistency. Especially since we most likely add an `oc` context as well. logs is for sure discussable, but at least versions and service should go into openshift. versions is specific to Minishift anyways. service also differs in the sense that we need to change the implementation of this command.

> What commands would not require openshift in the middle? just start/stop/delete

config, console, start, stop, delete, ip, docker-env, ssh, version

Anything which affects more the VM and Docker. Also as mentioned above oc related things, like the discussed 'oc env'. 'console' might arguably go under 'openshift' as well.

>  I keep looking into alignment with upstream, so minikube users has the easiest entry path into openshift

Sure, but alignment is only possible to a certain degree. After all our main goal is to make the usage of OpenShift easier. That partly conflicts with Minikube which has a different goal. Though similar, I would always expect there to be a difference in the concrete commands offered. In fact creating this openshift context could even add clarity. Commands which live outside this context can be considered shared between Qinikube and Minishift. They affect the basic life cycle commands for the VM + OpenShift/Kubernetes. The openshift context then contains commands specific to OpenShift. I actually see this as an advantage.

**Comment by jorgemoralespou** [comment]
As long as there is no commands in minikube that we move to the openshift context, you have a +1 from me.

**Comment by hferentschik** [comment]
Fair enough :-)

**Comment by praveenkumar** [comment]
@hferentschik below is the minikube commands and if we are agree on we shouldn't move any commands to openshift context which is part of minikube then we don't have anything left for this issue (version, logs and service is part of main minikube).

```
$ ./minikube -h
Minikube is a CLI tool that provisions and manages single-node Kubernetes clusters optimized for development workflows.

Usage:
  minikube [command]

Available Commands:
  addons           Modify minikube's kubernetes addons
  completion       Outputs minikube shell completion for the given shell (bash)
  config           Modify minikube config
  dashboard        Opens/displays the kubernetes dashboard URL for your local cluster
  delete           Deletes a local kubernetes cluster.
  docker-env       sets up docker env variables; similar to '$(docker-machine env)'
  get-k8s-versions Gets the list of available kubernetes versions available for minikube.
  ip               Retrieve the IP address of the running cluster.
  logs             Gets the logs of the running localkube instance, used for debugging minikube, not user code.
  service          Gets the kubernetes URL(s) for the specified service in your local cluster
  ssh              Log into or run a command on a machine with SSH; similar to 'docker-machine ssh'
  start            Starts a local kubernetes cluster.
  status           Gets the status of a local kubernetes cluster.
  stop             Stops a running local kubernetes cluster.
  version          Print the version of minikube.
```

**Comment by hferentschik** [comment]
@praveenkumar 

* _version_ of course needs to stay. It reports the Minishift version, what should more is _get_openshift_version_. A benefit of this is that once it moves into the _openshift_ context, we can get rid of the strange name and just call it _version_
* _logs_ also needs to stay. What I am proposing is to change the semantics a bit. _minishift logs_ shows the system logs of the VM, and we introduce a _minishift openshift logs_ which shows the OpenShift logs. I'd find that useful, but can live without
* _service_ this is the toughest one. Yes, there is an overlap here with Kubernetes, but in the end the semantics are different. Personally, I'd move it to _openshift_ since it will make the CLI more consistent and self explaining.

**Comment by jorgemoralespou** [comment]
+1 to all your suggestions @hferentschik

**Comment by praveenkumar** [comment]
> version of course needs to stay. It reports the Minishift version, what should more is get_openshift_version. A benefit of this is that once it moves into the openshift context, we can get rid of the strange name and just call it version

👍 

> logs also needs to stay. What I am proposing is to change the semantics a bit. minishift logs shows the system logs of the VM, and we introduce a minishift openshift logs which shows the OpenShift logs. I'd find that useful, but can live without

what do you mean by system logs here something like `dmesg`? we do need to keep in mind that all the iso should have same command to access those details.

> service this is the toughest one. Yes, there is an overlap here with Kubernetes, but in the end the semantics are different. Personally, I'd move it to openshift since it will make the CLI more consistent and self explaining.

I agree, we can move it to `openshift` context.

**Comment by praveenkumar** [comment]
Did a basic check from system logs side and looks like it's not consist across iso. So we are not going to touch `logs` command options.

For b2d iso

```
$ ls /var/log/
autologin        containers/      parallels.log    userdata.log
boot2docker.log  docker.log       udhcp.log        wtmp
```

For CentOS/RHEL iso

```
$ ls /var/log/
audit  btmp  containers  dmesg  lastlog  ppp  tallylog  tuned  userdata.log  wtmp  yum.log
```

**Comment by hferentschik** [comment]
Merged via pull request #500.
